api/pkg/s3/datastore/aws/aws.go

```diff
@@ -165,6 +165,26 @@ func (ad *AwsAdapter) DELETE(object *pb.DeleteObjectInput, ctx context.Context)
 	return NoError
 }
 
+func (ad *AwsAdapter) RestoreObject(object *pb.RestoreObjectInput, ctx context.Context) S3Error {
+
+	bucket := ad.backend.BucketName
+
+	newObjectKey := object.Bucket + "/" + object.Key
+
+	restoreInput := awss3.RestoreObjectInput{Bucket: &bucket, Key: &newObjectKey}
+
+	svc := awss3.New(ad.session)
+	_, err := svc.RestoreObject(&restoreInput)
+	if err != nil {
+		log.Logf("restore object failed, err:%v\n", err)
+		return InternalError
```

Comment:
why are we returning InternalError ? please return AWS specifc error, like this
`if err != nil {
    if aerr, ok := err.(awserr.Error); ok {
        switch aerr.Code() {
        case s3.ErrCodeObjectAlreadyInActiveTierError:
            fmt.Println(s3.ErrCodeObjectAlreadyInActiveTierError, aerr.Error())
        default:
            fmt.Println(aerr.Error())
        }
    } else {
        // Print the error, cast err to awserr.Error to get the Code and
        // Message from an error.
        fmt.Println(err.Error())
    }
    return
}`